File: Android/GameControllerSimulator/app2/src/main/java/cn/maoyanluo/gamecontrollersimulator2/generator/GamepadStateChange.kt
package cn.maoyanluo.gamecontrollersimulator2.generator

data class GamepadStateChange(
    val type: GamepadChangeType,

    val button: GamepadButton? = null,
    val buttonPressed: Boolean = false,

    val axis: GamepadAxis? = null,
    val axisValue: Short = 0,

    val trigger: GamepadTrigger? = null,
    val triggerValue: UByte = 0u
)

File: Windows/GameControllerSimulator/CommonLibrary/Generator/GamepadStateChange.cs
using System.Text.Json;

namespace CommonLibrary.Generator
{
    public class GamepadStateChange
    {
        public GamepadChangeType Type { get; set; }

        public GamepadButton? Button { get; set; }
        public bool ButtonPressed { get; set; }

        public GamepadAxis? Axis { get; set; }
        public short AxisValue { get; set; }

        public GamepadTrigger? Trigger { get; set; }
        public byte TriggerValue { get; set; }

        public override string ToString()
        {
            return JsonSerializer.Serialize(this);
        }

    }
}
